frame/support/src/traits/hooks.rs

```diff
@@ -189,7 +205,13 @@ pub trait IntegrityTest {
 }
 
 /// The pallet hooks trait. Implementing this lets you express some logic to execute.
-pub trait Hooks<BlockNumber> {
+///
+/// The `PreUpgradeState` generic type parameter is used by `pre_upgrade` to generated
+/// a state digest based on the pre-updrade state and will be passed to `post_upgrade`
+/// after upgrading for post-check. If there is no such need, `()` should be used.
+///
+/// TODO: use the `associated_type_defaults` feature once it is stable.
+pub trait Hooks<BlockNumber, PreUpgradeState: Default + Encode + Decode = ()> {
```

Comment:
Now that we decided to hardcode `Vec<u8>`, no need for this. 

```suggestion
pub trait Hooks<BlockNumber> {
```

Reply:
again, I really don't want to pollute the rest of the codebase for try-runtime.

Reply:
It is removed.